besu/src/test/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilderTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.controller;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.config.GenesisConfigFile;
+import org.hyperledger.besu.config.GenesisConfigOptions;
+import org.hyperledger.besu.config.StubGenesisConfigOptions;
+import org.hyperledger.besu.ethereum.core.PrivacyParameters;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolScheduleBuilder;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpecAdapters;
+import org.hyperledger.besu.evm.internal.EvmConfiguration;
+
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ConsensusScheduleBesuControllerBuilderTest {
+  @Mock private GenesisConfigFile genesisConfigFile;
+  @Mock private BesuControllerBuilder delegateBesuControllerBuilder1;
+  @Mock private BesuControllerBuilder delegateBesuControllerBuilder2;
+
+  @Test
+  public void createsCombinedProtocolScheduleWithMilestonesFromSingleProtocolSchedule() {
+    final StubGenesisConfigOptions genesisConfigOptions = new StubGenesisConfigOptions();
+    genesisConfigOptions.homesteadBlock(5L);
+    genesisConfigOptions.constantinopleBlock(10L);
+    final ProtocolSchedule protocolSchedule = createProtocolSchedule(genesisConfigOptions);
+
+    final Map<Long, BesuControllerBuilder> schedule = new HashMap<>();
```

Comment:
nit: for clarity, renaming this consensusSchedule would help distinguish from protocolSchedule

Reply:
done